### PR TITLE
feat(PromisesTable): filter by status

### PR DESCRIPTION
**Pull request by weiweitoo**
## Visual
- 'filter by status' text

## Functionalty
- if status filter is selected, filter promises
- ensure it can work in tandem with title search

### CODE REVIEW

**Comment by weiweitoo** [comment]
@nazreen Here is the PR. Still left 2 more items haven't done yet. :p
- ensure it can work in tandem with title search
- some sort of affordance when a status filter is selected

File: src/components/PromisesTable.vue [hunk 1]
```diff
@@ -79,14 +93,48 @@ export default {
...
+    }, {
+      value: 'Broken',
+      label: 'Broken'
+    }],
   }),
   methods: { formatDate },
   computed: {
     filteredPromises () {
-      return this.promises.filter(promise => {
+      var results = this.promises.filter(promise => {
```
Comment (nazreen) [inline src/components/PromisesTable.vue]: let's stick to ES6, so declare variables by either using `let` or `const`

File: src/views/List.vue [hunk 2]
```diff
@@ -44,7 +44,7 @@ export default {
         if (promise) {
           this.promises.push(promise)
         } else {
-          console.error(`promise with ID ${promiseID} is undefined`)
```
Comment (nazreen) [inline src/views/List.vue:47]: good spot! thanks!

File: src/components/PromisesTable.vue [hunk 3]
```diff
@@ -79,14 +93,48 @@ export default {
     exclude: { type: Array, default: () => [] }
   },
   data: () => ({
-    search: ''
+    search: '',
+    filterStatus : '',
+    statusOptions: [
```
Comment (nazreen) [inline src/components/PromisesTable.vue]: I think we can host this value in an external file

File: src/components/PromisesTable.vue [hunk 4]
```diff
@@ -79,14 +93,48 @@ export default {
...
   computed: {
     filteredPromises () {
-      return this.promises.filter(promise => {
+      var results = this.promises.filter(promise => {
         return promise.title.toLowerCase().includes(this.search.toLowerCase())
       })
+      results = results.filter(promise => {
+        return this.filterStatus == '' || promise.status == this.filterStatus;
+      })
+      return results;
```
Comment (nazreen) [inline src/components/PromisesTable.vue]: for this project repo, we standardise it to no semi-colons

File: src/components/PromisesTable.vue [hunk 5]
```diff
@@ -79,14 +93,48 @@ export default {
...
+    }],
   }),
   methods: { formatDate },
   computed: {
     filteredPromises () {
-      return this.promises.filter(promise => {
+      var results = this.promises.filter(promise => {
         return promise.title.toLowerCase().includes(this.search.toLowerCase())
       })
+      results = results.filter(promise => {
```
Comment (nazreen) [inline src/components/PromisesTable.vue]: best to define the filtering in its own method like how search is being done in the line above :)

**Review by nazreen — CHANGES_REQUESTED** [review]

File: src/components/PromisesTable.vue (hunk 1 above)
Comment (weiweitoo) [reply src/components/PromisesTable.vue]: Kay. No problems

File: src/components/PromisesTable.vue (hunk 3 above)
Comment (weiweitoo) [reply src/components/PromisesTable.vue]: Just realized we have a file that already hosting those value

File: src/components/PromisesTable.vue [hunk 6]
```diff
@@ -79,14 +99,20 @@ export default {
     exclude: { type: Array, default: () => [] }
   },
   data: () => ({
-    search: ''
+    search: '',
+    filterStatus : '',
+    statusOptions : ['All',...promiseStatusValues]
```
Comment (nazreen) [inline src/components/PromisesTable.vue]: @weiweitoo  I'm going to remove this 'All' value as when selected no promises would show as it's looking for `promise.status === 'All'`

the element-ui component for selecting promise status is actually clearable

**Review by nazreen — APPROVED** [review]

**Pull request merged**